### PR TITLE
[Stats Refresh] Insights Tags and Categories: add drop-down rows for groups

### LINKED ISSUE #10832: [Stats Refresh] Insights Tags and Categories: add drop-down rows for groups
On the Insights Tags and Categories card, if a post has multiple tags/categories, selecting the row should expand and show each individual tag/category. 

Design:
tags and categories amended

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #10832 

To test:

---
On a site that has a post with multiple tags/categories:
- Tip:  Hogwarts usually has some.
- Go to Stats > Insights > Tags and Categories.
- Verify:
  - The icon is `folderMultiple`.
  - The disclosure icon is pointed down.

tandc_init

---
Click on a row that has multiple tags/categories.
- Verify:
  - Individual rows appear for each tag/category.
    - Max of 10 "child" rows will be displayed.
    - NOTE: we don't currently have the totals for expanded rows. Ref https://github.com/wordpress-mobile/WordPress-iOS/issues/10945.
  - On the "parent" row, the disclosure icon is pointed up.
  - The row dividers above and below the expanded section are the full width of the row.
  - Selecting a "child" row displays a webview for that tag/category.

tandc_expanded

---
- Verify expanded rows persist on scrolling and rotation.
- Verify expanded rows are reset when Insights is first displayed.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@SylvesterWilmott - 

---
expanding_rows

---
tandc_init

---
tandc_expanded

---
ipad

**Comment by SylvesterWilmott** [comment]
This looks great @ScoutHarris !

I'm a little concerned by the jumpiness of the expanding/collapsing. In the .gif it appears that the cells above the one that expands/collapses is moving too which makes it a little difficult to follow what is happening. Is there a way to keep the other cells completely static and animate only the group that is expanding?

**Comment by ScoutHarris** [comment]
Hey @SylvesterWilmott . 

> jumpiness of the expanding/collapsing

Hrm. That's being caused by the need to redraw the card to resize it to accommodate adding/removing rows. Without the redraw, the rows were just smushed into the original card size. I'm not sure how else I can accomplish that. Do you mind if I add this to my todo list for later so can experiment and hopefully come up with a better solution?

**Comment by jklausa** [comment]
I also had similar issues with jumpiness in Plugins Detail view — I _think_ I fought with it for a long time, and didn't manage to fix it a 100% (though I have an idea how to do it now...), but maybe you could take a peek there for inspiration :P

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift [hunk 1]
```diff
@@ -183,8 +195,27 @@ extension SiteStatsInsightsTableViewController: SiteStatsInsightsDelegate {
     }
 
     func tabbedTotalsCellUpdated() {
-        tableView.beginUpdates()
-        tableView.endUpdates()
+        applyTableUpdates()
+    }
+
+    func expandedRowUpdated(_ row: StatsTotalRow) {
+        applyTableUpdates()
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift]: Sure, I can. But for here - everything after `applyTableUpdates` is just updating an array that has no effect on the table. So I didn't think waiting to `endUpdates` was necessary.

**Comment by ScoutHarris** [comment]
> maybe you could take a peek there for inspiration

@jklausa thank you!

**Pull request merged**